src/pydvl/influence/array.py

```diff
@@ -0,0 +1,390 @@
+        self,
+        nested_generators_of_tensors: Generator[
+            Generator[TensorType, None, None], None, None
+        ],
+    ):
+        """
+        Aggregates tensors from a generator of generators.
+
+        Implement this method to specify how tensors, nested in multiple layers of generators,
+        should be combined. Useful for complex data structures where tensors
+        are not directly accessible in a flat list.
+
+        Args:
+            nested_generators_of_tensors: A generator of generators, each yielding TensorType objects."""
+
+    @abstractmethod
+    def aggregate_from_generator(
+        self, tensor_generator: Generator[TensorType, None, None]
+    ):
+        """
+        Aggregates tensors from a single generator.
+
+        Implement this method to define how a sequence of tensors, provided by a generator,
+        should be combined. Suitable for straightforward scenarios where
+        tensors are yielded one by one.
+        """
+
+
+class ListAggregator(TensorAggregator):
+    """
+    An aggregator that collects tensors into list structures.
+    """
+
+    def aggregate_from_nested_generators(
+        self,
+        nested_generators_of_tensors: Generator[
+            Generator[TensorType, None, None], None, None
+        ],
+    ):
+        """
+         Aggregates tensors from a nested generator structure into a list of lists. Each inner generator
+         is converted into a list of tensors, resulting in a nested list structure.
+
+         Args:
+             nested_generators_of_tensors: A generator of generators, where each inner generator yields
+                 TensorType objects.
+
+        Returns:
+            A list of lists, where each inner list contains tensors returned from one of the inner generators.
+        """
+        return [list(tensor_gen) for tensor_gen in nested_generators_of_tensors]
+
+    def aggregate_from_generator(
+        self, tensor_generator: Generator[TensorType, None, None]
+    ):
+        """
+        Aggregates tensors from a single-level generator into a list. This method simply collects
+        each tensor emitted by the generator into a single list.
+
+        Args:
+            tensor_generator: A generator that yields TensorType objects.
+
+        Returns:
+            A list containing all the tensors provided by the tensor_generator.
+        """
+        return [t for t in tensor_generator]
+
+
+class LazyArrayProvider(ABC):
+    """
+    Abstract base class representing a provider of lazily evaluated, array-like data. This class
+    is tailored for situations where data is not immediately computed or loaded, but is instead
+    generated or processed on-demand. The 'array-like' aspect refers to the data structure or
+    format rather than the ability to perform typical array operations like indexing.
+    """
+
+    @abstractmethod
+    def compute(self, tensor_aggregator: Optional[TensorAggregator] = None):
```

Comment:
I think this is another example of an IMO unnecessary type prefix. It's an aggregator of type `TensorAggregator`. As a user, I don't need to see a name `tensor_aggregator`, that's just only going to take up space.